akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -100,23 +178,71 @@ final case class HttpCookie(
   def getMaxAge: OptionalLong = maxAge.asPrimitive
   /** Java API */
   def getExpires: Optional[jm.DateTime] = expires.map(_.asJava).asJava
+
+  def withName(name: String): HttpCookie = copy(name = name)
+  def withValue(value: String): HttpCookie = copy(value = value)
+  /** Scala API */
+  def withExpires(dateTime: DateTime): HttpCookie = copy(expires = Some(dateTime))
   /** Java API */
-  def withExpires(dateTime: jm.DateTime): headers.HttpCookie = copy(expires = Some(dateTime.asScala))
-  /** Java API */
-  def withDomain(domain: String): headers.HttpCookie = copy(domain = Some(domain))
-  /** Java API */
-  def withPath(path: String): headers.HttpCookie = copy(path = Some(path))
-  /** Java API */
-  def withMaxAge(maxAge: Long): headers.HttpCookie = copy(maxAge = Some(maxAge))
-  /** Java API */
-  def withSecure(secure: Boolean): headers.HttpCookie = copy(secure = secure)
-  /** Java API */
-  def withHttpOnly(httpOnly: Boolean): headers.HttpCookie = copy(httpOnly = httpOnly)
+  def withExpires(dateTime: jm.DateTime): HttpCookie = copy(expires = Some(dateTime.asScala))
+
+  def withDomain(domain: String): HttpCookie = copy(domain = Some(domain))
+  def withPath(path: String): HttpCookie = copy(path = Some(path))
+  def withMaxAge(maxAge: Long): HttpCookie = copy(maxAge = Some(maxAge))
+  def withSecure(secure: Boolean): HttpCookie = copy(secure = secure)
+  def withHttpOnly(httpOnly: Boolean): HttpCookie = copy(httpOnly = httpOnly)
+  def withSameSite(sameSite: jm.headers.SameSite): HttpCookie = copy(sameSite = Option(sameSite.asScala()))
```

Comment:
Ok, this one should also work for the Scala one, I guess.

Reply:
Ah, no it doesn't, so should we add another overload for the Scala API (without the Option)?